operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/AbstractDependentResource.java

```diff
@@ -0,0 +1,33 @@
+package io.javaoperatorsdk.operator.api.reconciler.dependent;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.javaoperatorsdk.operator.api.reconciler.Context;
+
+public abstract class AbstractDependentResource<R, P extends HasMetadata>
+    implements DependentResource<R, P> {
+
+  @Override
+  public void reconcile(P primary, Context context) {
+    var actual = getResource(primary);
+    var desired = desired(primary, context);
+    if (actual.isEmpty()) {
+      create(desired, context);
+    } else {
+      if (!match(actual.get(), desired, context)) {
+        update(actual.get(), desired, context);
+      }
+    }
+  }
+
+  protected abstract R desired(P primary, Context context);
+
+  protected boolean match(R actual, R target, Context context) {
+    return actual.equals(target);
```

Comment:
This default implementation will result in infinite loops when working with Kubernetes resources.

Reply:
This is changed since. Please pull.